Common/Tools/TrackTuner.h

```diff
@@ -476,38 +493,72 @@ struct TrackTuner : o2::framework::ConfigurableGroup {
+    grDcaZPullVsPtPionData.reserve(nPhiBins);
+
+    /// Lambda expression to get the TGraphErrors from file
+    auto loadGraph = [&](int phiBin, const std::string& strBaseName) -> std::unique_ptr<TGraphErrors> {
+      std::string strGraphName = inputNphiBins != 0 ? fmt::format("{}_{}", strBaseName, phiBin) : strBaseName;
+      TObject* obj = td->Get(strGraphName.c_str());
+      if (!obj) {
+        LOG(fatal) << "[TrackTuner]     TGraphErrors not found in the Input Root file: " << strGraphName;
+        td->ls();
+        return nullptr;
+      }
+      return std::unique_ptr<TGraphErrors>(dynamic_cast<TGraphErrors*>(obj));
+    };
+
+    if (inputNphiBins != 0) {
+      LOG(info) << "[TrackTuner]    Loading phi-dependent XY TGraphErrors";
     }
+    for (int iphibin = 0; iphibin < nPhiBins; ++iphibin) {
 
-    std::string grDcaZResNameMC = "resCurrentDcaZ";
-    std::string grDcaZMeanNameMC = "meanCurrentDcaZ";
-    std::string grDcaZPullNameMC = "pullsCurrentDcaZ";
-    std::string grDcaZResNameData = "resUpgrDcaZ";
-    std::string grDcaZMeanNameData = "meanUpgrDcaZ";
-    std::string grDcaZPullNameData = "pullsUpgrDcaZ";
-
-    grDcaZResVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZResNameMC.c_str())));
-    grDcaZResVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZResNameData.c_str())));
-    grDcaZMeanVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZMeanNameMC.c_str())));
-    grDcaZMeanVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZMeanNameData.c_str())));
-    grDcaZPullVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZPullNameMC.c_str())));
-    grDcaZPullVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZPullNameData.c_str())));
-    if (!grDcaZResVsPtPionMC.get() || !grDcaZResVsPtPionData.get() || !grDcaZMeanVsPtPionMC.get() || !grDcaZMeanVsPtPionData.get() || !grDcaZPullVsPtPionMC.get() || !grDcaZPullVsPtPionData.get()) {
-      LOG(fatal) << "Something wrong with the names of the correction graphs for dcaZ. Fix it!";
+      grDcaXYResVsPtPionMC[iphibin].reset(loadGraph(iphibin, "resCurrentDcaXY").release());
+      grDcaXYResVsPtPionData[iphibin].reset(loadGraph(iphibin, "resUpgrDcaXY").release());
+      grDcaXYMeanVsPtPionMC[iphibin].reset(loadGraph(iphibin, "meanCurrentDcaXY").release());
+      grDcaXYMeanVsPtPionData[iphibin].reset(loadGraph(iphibin, "meanUpgrDcaXY").release());
+      grDcaXYPullVsPtPionMC[iphibin].reset(loadGraph(iphibin, "pullsCurrentDcaXY").release());
+      grDcaXYPullVsPtPionData[iphibin].reset(loadGraph(iphibin, "pullsUpgrDcaXY").release());
+
+      if (!grDcaXYResVsPtPionMC[iphibin].get() || !grDcaXYResVsPtPionData[iphibin].get() || !grDcaXYMeanVsPtPionMC[iphibin].get() || !grDcaXYMeanVsPtPionData[iphibin].get() || !grDcaXYPullVsPtPionMC[iphibin].get() || !grDcaXYPullVsPtPionData[iphibin].get()) {
+        LOG(fatal) << "[TrackTuner]     Something wrong with the names of the correction graphs for dcaXY. Fix it!";
```

Comment:
I'd add here in the message of the `fatal` a reference to the `iPhiBin` that causes the issue